### PR TITLE
Feature/idcom 1996 update function implementation

### PR DESCRIPTION
**Pull request by frank-haomeng-wang**
I have completed the update function.

### CODE REVIEW

File: program_v2/.program_client/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[package]
```
Comment (rado0x54) [inline program_v2/.program_client/Cargo.toml]: Remove this file

File: program_v2/.program_client/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// DO NOT EDIT - automatically generated file
```
Comment (rado0x54) [inline program_v2/.program_client/src/lib.rs]: Remove?

File: program_v2/programs/sol-did/src/instructions/update.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ctx: Context<Update>,
+    mut update_information: UpdateStruct,
+    eth_signature: Option<Secp256k1RawSignature>,
+) -> Result<()> {
+    let data = &mut ctx.accounts.did_data;
+    data.services = update_information.services;
+    data.native_controllers = update_information.native_controllers;
+    data.other_controllers = update_information.other_controllers;
+    data.initial_verification_method.flags = 0;
+    let default_alas = & data.initial_verification_method.alias;
```
Comment (rado0x54) [inline program_v2/programs/sol-did/src/instructions/update.rs]: Spelling of alas

File: program_v2/programs/sol-did/src/instructions/update.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    data.services = update_information.services;
+    data.native_controllers = update_information.native_controllers;
+    data.other_controllers = update_information.other_controllers;
+    data.initial_verification_method.flags = 0;
+    let default_alas = &data.initial_verification_method.alias;
+    let new_methods = &mut update_information.verification_methods;
+
+    for i in new_methods.clone() {
+        if i.alias == *default_alas {
+            new_methods.retain(|x| x.alias == *default_alas);
```
Comment (rado0x54) [inline program_v2/programs/sol-did/src/instructions/update.rs:19]: For and retain are both O(n). Can you implement this by only running through the list once.

File: program_v2/programs/sol-did/src/instructions/update.rs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            data.initial_verification_method.flags = i.flags;
+            break;
```
Comment (rado0x54) [inline program_v2/programs/sol-did/src/instructions/update.rs:21]: The case where default cannot be found is missing.

**Review by rado0x54 — CHANGES_REQUESTED** [review]

File: program_v2/programs/sol-did/src/instructions/update.rs (hunk 5 above)
Comment (frank-haomeng-wang) [reply program_v2/programs/sol-did/src/instructions/update.rs:21]: if it can not be found, the flag is just 0 as I set above

**Pull request closed without merging**